File: backend/app/schemas/guidance.py
"""
Pydantic schemas for Phase 4 personalized patient guidance
and caregiver support guidance.
"""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import AiSummaryStatus, GuidanceType
from app.schemas.common import ORMBaseSchema


class PersonalizedGuidanceContent(BaseModel):
    """Strict shape for personalized patient guidance."""

    model_config = ConfigDict(extra="forbid")

    overview: str | None = None

    immediate_actions: list[str] = Field(default_factory=list)
    questions_for_doctor: list[str] = Field(default_factory=list)
    medication_guidance: list[str] = Field(default_factory=list)
    nutrition_guidance: list[str] = Field(default_factory=list)
    activity_guidance: list[str] = Field(default_factory=list)
    warning_signs: list[str] = Field(default_factory=list)
    follow_up_plan: list[str] = Field(default_factory=list)
    emotional_support: list[str] = Field(default_factory=list)

    disclaimer: str


class CaregiverGuidanceContent(BaseModel):
    """Strict shape for caregiver support guidance."""

    model_config = ConfigDict(extra="forbid")

    overview: str | None = None

    daily_support: list[str] = Field(default_factory=list)
    medication_support: list[str] = Field(default_factory=list)
    appointment_support: list[str] = Field(default_factory=list)
    nutrition_support: list[str] = Field(default_factory=list)
    mobility_support: list[str] = Field(default_factory=list)
    emotional_support: list[str] = Field(default_factory=list)
    warning_signs: list[str] = Field(default_factory=list)
    caregiver_wellbeing: list[str] = Field(default_factory=list)

    disclaimer: str


class GuidanceBaseReadSchema(ORMBaseSchema):
    """Common database fields returned for both guidance types."""

    id: UUID
    report_id: UUID
    patient_id: UUID
    analysis_id: UUID
    guidance_type: GuidanceType
    status: AiSummaryStatus
    error_message: str | None
    model_name: str | None
    created_at: datetime
    updated_at: datetime


class PersonalizedGuidanceReadSchema(GuidanceBaseReadSchema):
    """Response schema for personalized patient guidance."""

    content: PersonalizedGuidanceContent | None = None


class CaregiverGuidanceReadSchema(GuidanceBaseReadSchema):
    """Response schema for caregiver guidance."""

    content: CaregiverGuidanceContent | None = None

File: frontend/src/types/guidance.ts
/**
 * Mirrors the backend's `app.schemas.guidance` contracts (Phase 4 —
 * Personalized Guidance and Caregiver Support agents).
 */

import type { AiSummaryStatus } from "@/types/report";

export type GuidanceType = "patient_guidance" | "caregiver_guidance";

export interface PersonalizedGuidanceContent {
  precautions: string[];
  nutrition_guidance: string[];
  lifestyle_guidance: string[];
  questions_for_doctor: string[];
  follow_up_checklist: string[];
  appointment_preparation: string[];
  warning_signs: string[];
  limitations: string[];
  disclaimer: string;
}

export interface CaregiverGuidanceContent {
  daily_support: string[];
  emotional_support: string[];
  appointment_support: string[];
  medication_support: string[];
  nutrition_and_hydration: string[];
  fatigue_and_comfort_support: string[];
  symptoms_to_observe: string[];
  emergency_warning_signs: string[];
  caregiver_self_care: string[];
  questions_for_medical_team: string[];
  limitations: string[];
  disclaimer: string;
}

interface GuidanceBase {
  id: string;
  report_id: string;
  patient_id: string;
  analysis_id: string;
  guidance_type: GuidanceType;
  status: AiSummaryStatus;
  error_message: string | null;
  model_name: string | null;
  created_at: string;
  updated_at: string;
}

export interface PersonalizedGuidance extends GuidanceBase {
  content: PersonalizedGuidanceContent | null;
}

export interface CaregiverGuidance extends GuidanceBase {
  content: CaregiverGuidanceContent | null;
}
